superset/connectors/sqla/models.py

```diff
@@ -1454,6 +1454,11 @@ def has_extra_cache_key_calls(self, query_obj: QueryObjectDict) -> bool:
             templatable_statements.append(extras["where"])
         if "having" in extras:
             templatable_statements.append(extras["having"])
+        # make sure the RLS filters are included
+        if config["ENABLE_ROW_LEVEL_SECURITY"] and self.is_rls_supported:
+            templatable_statements += [
+                f.clause
+                for f in security_manager.get_rls_filters(self) ]
```

Comment:
Please make sure to run black on your code prior to opening a PR (check `pre-commit` hooks in [CONTIBUTING.md](https://github.com/apache/incubator-superset/blob/master/CONTRIBUTING.md#git-hooks))
```suggestion
        if config["ENABLE_ROW_LEVEL_SECURITY"] and self.is_rls_supported:
            templatable_statements += [
                f.clause for f in security_manager.get_rls_filters(self)
            ]
```